.buildkite/pipeline.yml

```diff
@@ -1,3 +1,66 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+
+env:
+  SETUP_GVM_VERSION: 'v0.5.0'
+  DOCKER_COMPOSE_VERSION: '1.25.5'
+  ELASTIC_PACKAGE_COMPOSE_DISABLE_ANSI: "true"
+  KIND_VERSION: 'v0.17.0'
+  K8S_VERSION: 'v1.26.0'
+
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - group: "Check and build"
+    key: "tests"
+    steps:
+      - label: ":go: Run check-ci"
+        key: check-ci
+        command: "make check-ci"
+        agents:
+          image: "golang:1.20.3"
+          cpu: "8"
+          memory: "4G"
+
+      - label: ":go: Run local build"
+        key: make-local
+        command: "make local"
+        agents:
+          image: "golang:1.20.3"
+          cpu: "8"
+          memory: "4G"
+
+  - group: "Run tests"
+    steps:
+      - label: "Run unit test"
+        key: unit-test
+        commands:
+          - "chmod +x .buildkite/scripts/setenv.sh"
+          - ".buildkite/scripts/setenv.sh with-go"
+          - "make test-unit"
+          - "make junit-report"
+          - "buildkite-agent artifact upload build/*.xml"
+        agents:
+          provider: "gcp"
+
+
+      - label: "Run integration tests"
+        key: int-test
+        commands:
+          - "chmod +x .buildkite/scripts/setenv.sh"
+          - ".buildkite/scripts/setenv.sh with-go-docker-compose"
+          - "make test-int"
+          - "make junit-report"
+          - "buildkite-agent artifact upload build/*.xml"
+        agents:
+          provider: "gcp"
+
+      - label: ":junit: Junit annotate"
+        commands:
+          - "buildkite-agent artifact download 'build/*.xml' build/"
```

Comment:
Is it necessary to add this command ? At least with elastic-package it was not needed. The plugin download the files/artifacts specified with the key `artifacts`.

Reply:
let me check it, I've changed it, but haven't commited yet.

Reply:
Yes, you are right, mentioned you solution is also working properly, let's use your way because of the same approach.
Thank you!

Reply:
I see that the other steps have been updated including the `artifact_paths` key :+1: 

I think it is just missing to remove the command key in the junit step, as in the example: https://github.com/buildkite-plugins/junit-annotate-buildkite-plugin#example

```suggestion
```

Reply:
Is that command in this step needed @sharbuz ?

Reply:
Let me check it...